### PR TITLE
Fix leather horse armor being unequipable

### LINKED ISSUE #7140: Horse armor equip issue
### Skript/Server Version

```markdown
[17:25:18 INFO]: [Skript] Skript's aliases can be found here: https://github.com/SkriptLang/skript-aliases
[17:25:18 INFO]: [Skript] Skript's documentation can be found here: https://docs.skriptlang.org/
[17:25:18 INFO]: [Skript] Skript's tutorials can be found here: https://docs.skriptlang.org/tutorials
[17:25:18 INFO]: [Skript] Server Version: 1.21.1-DEV-077c3a9 (MC: 1.21.1)
[17:25:18 INFO]: [Skript] Skript Version: 2.9.2 (skriptlang-github)
[17:25:18 INFO]: [Skript] Installed Skript Addons: 
[17:25:18 INFO]: [Skript]  - skript-placeholders v1.6.0 (https://github.com/APickledWalrus/skript-placeholders)
[17:25:18 INFO]: [Skript]  - skript-gui v1.3 (https://github.com/APickledWalrus/skript-gui)
[17:25:18 INFO]: [Skript]  - skript-reflect v2.4 (https://github.com/SkriptLang/skript-reflect)
[17:25:18 INFO]: [Skript]  - SkQuery v4.1.10
[17:25:18 INFO]: [Skript]  - SkBee v3.6.1 (https://github.com/ShaneBeee/SkBee)
[17:25:18 INFO]: [Skript] Installed dependencies: 
[17:25:18 INFO]: [Skript]  - Vault v1.7.3-b131
[17:25:18 INFO]: [Skript]  - WorldGuard v7.0.12+829a4a4
```

### Bug Description

Horses cannot be equipped with leather horse armor with the equip effect. All other horse armors work, however Material.LEATHER_HORSE_ARMOR is missing [here](https://github.com/SkriptLang/Skript/blob/ba32d7ad48edb8e33d5f5d2e44d03bba94ffb2de/src/main/java/ch/njol/skript/effects/EffEquip.java#L102).

### Expected Behavior

The horse should have the leather horse armor equipped.

### Steps to Reproduce

Summon a horse, and attempt to equip it with horse armor.

`summon horse at player`
`equip last spawned horse with leather horse armor`

### Errors or Screenshots

_No response_

### Other

_No response_

### Agreement

- [X] I have read the guidelines above and affirm I am following them with this report.

### PR DESCRIPTION
**Pull request by Efnilite**
### Description
Fixes #7140.

---
**Target Minecraft Versions:** any 
**Requirements:** none 
**Related Issues:** none

### CODE REVIEW

**Comment by cheeezburga** [comment]
Leather horse armour was only added in 1.14, and I think the 1.20.6+ test is resolved by #6849?

**Comment by Efnilite** [comment]
Tests failing due to a 1.20.6-only Paper bug (https://github.com/PaperMC/Paper/pull/11139).

**Review by cheeezburga — APPROVED** [review]

File: src/main/java/ch/njol/skript/effects/EffEquip.java [hunk 1]
```diff
@@ -47,26 +12,47 @@
...
+	private static ItemType HORSE_ARMOR;
+
 	static {
+		try {
+			HORSE_ARMOR = new ItemType(Material.IRON_HORSE_ARMOR, Material.GOLDEN_HORSE_ARMOR,
+				Material.DIAMOND_HORSE_ARMOR, Material.LEATHER_HORSE_ARMOR);
+		} catch (NoSuchFieldError ex) {
+			HORSE_ARMOR = new ItemType(Material.IRON_HORSE_ARMOR, Material.GOLDEN_HORSE_ARMOR,
+				Material.DIAMOND_HORSE_ARMOR);
+		}
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/effects/EffEquip.java:49]: why isn't this a version check

File: src/main/java/ch/njol/skript/effects/EffEquip.java [hunk 2]
```diff
@@ -47,26 +12,47 @@
...
+
 		Skript.registerEffect(EffEquip.class,
 				"equip [%livingentities%] with %itemtypes%",
 				"make %livingentities% wear %itemtypes%",
 				"unequip %itemtypes% [from %livingentities%]",
-				"unequip %livingentities%'[s] (armor|equipment)"
-			);
+				"unequip %livingentities%'[s] (armor|equipment)");
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/effects/EffEquip.java:55]:
```suggestion
				"unequip %livingentities%'[s] (armo[u]r|equipment)");
```
while we're here

**Review by sovdeeth — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/effects/EffEquip.java [hunk 3]
```diff
@@ -47,26 +12,47 @@
...
 })
 @Since("1.0, 2.7 (multiple entities, unequip)")
 public class EffEquip extends Effect {
 
+	private static final ItemType HORSE_ARMOR;
+
 	static {
+		if (Skript.isRunningMinecraft(1, 14)) {
+			HORSE_ARMOR = new ItemType(Material.IRON_HORSE_ARMOR, Material.GOLDEN_HORSE_ARMOR,
+				Material.DIAMOND_HORSE_ARMOR, Material.LEATHER_HORSE_ARMOR);
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/effects/EffEquip.java:45]: needs 2 tabs

File: src/main/java/ch/njol/skript/effects/EffEquip.java [hunk 4]
```diff
@@ -47,26 +12,47 @@
...
+		} else {
+			HORSE_ARMOR = new ItemType(Material.IRON_HORSE_ARMOR, Material.GOLDEN_HORSE_ARMOR,
+				Material.DIAMOND_HORSE_ARMOR);
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/effects/EffEquip.java:48]: 2 tabs

**Review by sovdeeth — APPROVED** [review]

**Review by Moderocky — APPROVED** [review]

**Pull request merged**